src/Microsoft.AspNet.OData/EnableQueryAttribute.cs

```diff
@@ -71,6 +73,37 @@ public override void OnActionExecuted(HttpActionExecutedContext actionExecutedCo
                 throw Error.Argument("actionExecutedContext", SRResources.ActionExecutedContextMustHaveActionContext);
             }
 
+            bool hasExpandQuery = false;
+            string queryString = request.RequestUri.Query;
+            var absoluteUri = request.RequestUri.AbsoluteUri;
+            bool hasEmptyQueryString = string.IsNullOrEmpty(queryString);
+            string baseUriWithPath = hasEmptyQueryString ? absoluteUri : absoluteUri.Substring(0, absoluteUri.IndexOf('?', 0));
+
+            if (!hasEmptyQueryString)
+            {
+                var queries = queryString.Split('&');
+
+                hasExpandQuery = queries.Any(x =>
+                {
+                    var xToUpperInvariant = x.ToUpperInvariant();
```

Comment:
Why do we need to make allocate an `UpperInvariant` copy of the string if we're using `StringComparison.OrdinalIgnoreCase` for the comparison?

Reply:
Fixed